File: fixtures/scale/scala/large/src/cryptobench/ciphers/beaufort/BeaufortKey.scala
package cryptobench.ciphers.beaufort

/** Parameters for the beaufort cipher. */
final case class BeaufortKey(keyword: String) {
  def keyCharAt(position: Int): Char = keyword.charAt(position % keyword.length)
}

object BeaufortKey {
  def default(): BeaufortKey = BeaufortKey("GRANITE")
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/beaufort/BeaufortKey.java
package cryptobench.ciphers.beaufort;

public class BeaufortKey {
    private final String keyword;

    public BeaufortKey(String keyword) {
        this.keyword = keyword.toUpperCase();
    }

    public String getKeyword() {
        return keyword;
    }

    public char keyCharAt(int position) {
        return keyword.charAt(position % keyword.length());
    }

    public static BeaufortKey defaultKey() {
        return new BeaufortKey("GRANITE");
    }
}
